build.gradle

```diff
@@ -1166,10 +1281,11 @@ project(':iceberg-spark3-runtime') {
   }
 
   dependencies {
-    compile project(':iceberg-spark3')
-    compile project(':iceberg-spark3-extensions')
-    compile project(':iceberg-aws')
-    compile(project(':iceberg-nessie')) {
+    implementation project(':iceberg-api')
```

Comment:
I think this should be an `api` dependency here and most other places.

Reply:
sgtm